Scan first product

PrajwalM2212 opened the pull request:
## Description

If the history is empty, a "Scan first product" button is added.

## Related issues and discussion
#166 
 
 ## Screen-shots, if any
![scanfirst](https://user-images.githubusercontent.com/30733262/36676174-f554469e-1b30-11e8-848d-3ab96f912ab5.png)

Shubham2320 commented:
I think one image or a textView with the same message should also appear. At present, the layout seems quite empty.

PrajwalM2212 commented:
@Shubham2320  The issue wanted only a button . Let us see what @teolemon has to say .

Shubham2320 commented:
yeah, that can be added later. :)

teolemon commented:
Feel free:
-to ask for clarifications before starting coding
- to be creative, and do something smarter than my boring suggestions :-)

More seriously, this is a good start, but it indeed looks a bit empty. I guess we could use the blue and white button design we use in the rest of the app.

We could also add a descriptive text above the button like: "All the products you scan or search for will appear here. Swipe left if you want to remove a product. Your history is yours: it is never sent online, and you can easily export it to use it outside Open Food Facts. "

jaztriumph commented:
I think we should take away the 'no data' toast and manage to show in a text view itself because I think toast don't look good.

PrajwalM2212 commented:
@jaztriumph  Thanks for the suggestion .

PrajwalM2212 commented:
![empty_history](https://user-images.githubusercontent.com/30733262/36702914-275d0c4a-1b7f-11e8-9ff6-8d49ea2b603f.png)

Would this be okay ?

teolemon commented:
Can you remove the yellow background make the text gray ? For the "Scan first product" can you reuse the same style as the Remove allergen buttons in the allergen view ?

Karljoones commented:
I would second the changes that @teolemon mentioned, however, it might be best to change the text. Maybe remove the ALL CAPS? And rephrase:

> Your viewed product history will be listed here! 
> This history is for your eyes only and is stored locally.

PrajwalM2212 commented:
I will do the changes.

PrajwalM2212 commented:
![scan](https://user-images.githubusercontent.com/30733262/36727538-a66d5b3e-1be3-11e8-80e5-ea6615084a74.png)

PrajwalM2212 commented:
@teolemon @Karljoones Made the changes.

Karljoones commented:
@prajwalm2212 much better! A couple of final things though. Can you use center justification with the text? And maybe a little bit more padding between the two and finally, the text just needs to be a little greyer to more blend into the background!

teolemon commented:
Also punctuation at the end of sentences. :)

PrajwalM2212 commented:
![scan](https://user-images.githubusercontent.com/30733262/36730233-ae4cd06a-1bec-11e8-9e48-914c830a885f.png)
@Karljoones  The text was already center justified . I have added more padding . I have made the text greyer as you requested .
@teolemon I have added punctuations at the end of sentences.

Karljoones commented:
Hi, the text is centered but the justification of the text is not. Off the top of my head, if you use android:gravity="center" might resolve it.

PrajwalM2212 commented:
![scan](https://user-images.githubusercontent.com/30733262/36792484-b957cf26-1cc0-11e8-848e-eb829378fae1.png)
@Karljoones  I think this is what you were looking for.

teolemon commented on app/src/main/java/openfoodfacts/github/scrachx/openfood/views/HistoryScanActivity.java:
```diff
@@ -290,4 +312,39 @@ public void onDraw(Canvas c, RecyclerView parent, RecyclerView.State state) {
+
+    public void setInfo(TextView view) {
+
+        String info = "     Your viewed product history will be listed here.\n" +
```
Are the spaces normal ?

PrajwalM2212 replied:
@teolemon To get the text centered justified , I had to put some spaces . I had already used android:gravity="center" but that did not solve the issue .

PrajwalM2212 replied:
@teolemon  I have solved it now . Thanks to suggestion from @Karljoones

PrajwalM2212 commented:
@teolemon 
It is now centre justified on smaller devices also .
![scan](https://user-images.githubusercontent.com/30733262/36794776-4bbb4e3c-1cc7-11e8-99ea-dd49ffb026f9.png)

Karljoones commented:
Thanks for making the changes @PrajwalM2212!
